src/main/java/org/sagebionetworks/web/client/widget/table/modal/fileview/TableType.java

```diff
@@ -2,72 +2,89 @@
 
 // Table, or a Project View, or a combination View composed of Files/Folders/Tables
 
+import static org.sagebionetworks.web.shared.WebConstants.DATASET;
+import static org.sagebionetworks.web.shared.WebConstants.DOCKER;
 import static org.sagebionetworks.web.shared.WebConstants.FILE;
 import static org.sagebionetworks.web.shared.WebConstants.FOLDER;
 import static org.sagebionetworks.web.shared.WebConstants.PROJECT;
+import static org.sagebionetworks.web.shared.WebConstants.SUBMISSION_VIEW;
 import static org.sagebionetworks.web.shared.WebConstants.TABLE;
+import static org.sagebionetworks.web.shared.WebConstants.VIEW;
+
+import java.util.Objects;
 
 import org.sagebionetworks.repo.model.Entity;
 import org.sagebionetworks.repo.model.table.Dataset;
 import org.sagebionetworks.repo.model.table.EntityView;
 import org.sagebionetworks.repo.model.table.SubmissionView;
+import org.sagebionetworks.repo.model.table.Table;
 import org.sagebionetworks.repo.model.table.TableEntity;
 import org.sagebionetworks.repo.model.table.ViewTypeMask;
 import org.sagebionetworks.web.client.DisplayConstants;
 
 
-public enum TableType {
-	table(null, DisplayConstants.TABLE),
-	submission_view(null, DisplayConstants.SUBMISSION_VIEW),
-	projects(PROJECT, DisplayConstants.PROJECT_VIEW),
-	tables(TABLE, DisplayConstants.VIEW),
-	folders(FOLDER, DisplayConstants.VIEW),
-	folders_tables(FOLDER | TABLE, DisplayConstants.VIEW),
-	files(FILE, DisplayConstants.FILE_VIEW),
-	files_tables(FILE | TABLE, DisplayConstants.VIEW),
-	files_folders(FILE | FOLDER, DisplayConstants.VIEW),
-	files_folders_tables(FILE | FOLDER | TABLE, DisplayConstants.VIEW),
-	dataset(FILE, DisplayConstants.DATASET);
+public class TableType {
```

Comment:
TableType is no longer an enum and is now a 2-tuple object consisting of the table class and the view type mask